src/app.rs

```diff
@@ -648,6 +653,24 @@ impl App {
         }
     }
 
+    pub fn copy_song_url(&mut self) {
+        let clipboard = match &mut self.clipboard_context {
```

Comment:
Looks like this could be using an `if let`?

```rust
if let Some(clipboard_context) = &mut self.clipboard_context {
  ...
}
```

Reply:
I used an "early return" here to save an indentation level, do you want me to change it?:)